### ISSUE TITLE
bug: Staging competition endpoint reporting wrong info about the winner

### ISSUE THREAD

**Issue by harisang** [cowprotocol/services#1956]
### Problem
Barn-1inch solver executing this settlement, that contains 2 orders:
[0x7783c5bc05c20e5556d6b015a1788f38105955a0b16c70c793cca5513b7a73b7](https://etherscan.io/tx/0x7783c5bc05c20e5556d6b015a1788f38105955a0b16c70c793cca5513b7a73b7)

[Competition endpoint](https://barn.api.cow.fi/mainnet/api/v1/solver_competition/by_tx_hash/0x7783c5bc05c20e5556d6b015a1788f38105955a0b16c70c793cca5513b7a73b7) reports that the winner contained only one solution.

**Comment by sunce86** [comment]
This is a one time error due to switching to colocation. What happened:

1. At [13:11](https://production-6de61f.kb.eu-central-1.aws.cloud.es.io/app/r/s/LoEuM) we were running non-colocated driver and auction `9073831`, but did not end up with mined transaction, however, we did save `<auction_id, <from, nonce>> = <9073831, <1inch, X>>` pair to database.
2. We switched to colocated driver.
3. Colocated driver runs some other auction, and 1inch wins the competition and tx `0x7783c5bc05c20e5556d6b015a1788f38105955a0b16c70c793cca5513b7a73b7` gets mined, with the `<from, nonce> = <1inch, X` like in the step (1). 
4. on_settlement_event_updater sees the tx `0x7783c5bc05c20e5556d6b015a1788f38105955a0b16c70c793cca5513b7a73b7` and uses <1inch, nonce> to find the auction id, but it finds the one from the non-colocated driver, because that one has the priority in the [code](https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/on_settlement_event_updater.rs#L116-L127).

Bottom line, we end up with a wrong <auction_id, transaction_hash> association for this single competition that happened just before the colocation was activated.

@MartinquaXD we should be able to fix this just by erasing the https://github.com/cowprotocol/services/blob/main/crates/autopilot/src/on_settlement_event_updater.rs#L116-L119 
WDYT?

**Comment by harisang** [comment]
Thanks for investigating, Dusan!
